### ISSUE TITLE
Chunks are sent in disconnected state

### ISSUE THREAD

**Issue by dbrgn** [threema-ch/threema-web#619]
Here's a debug log:

```
[Debug] [StatusController] – "State change:" – "ok" – "->" – "error" (angular.js, line 15504)
[Debug] Sending thumbnail request for – "contact" – "z1f2qNrBA6U=" (angular.js, line 15504)
[Debug] Sending – "request/thumbnail" – "message" (angular.js, line 15504)
[Error] Cannot send task message in "closed" state
https://web-beta.threema.ch/node_modules/@saltyrtc/client/dist/saltyrtc-client.es5.js?v=2.1.0-beta.6:321
    (anonymous function) (angular.js:11814)
    (anonymous function) (angular.js:21558)
    completeTask (angular.js:21193)
    (anonymous function) (angular.js:6789)
[Error] Cannot send task message in "closed" state
https://web-beta.threema.ch/node_modules/@saltyrtc/client/dist/saltyrtc-client.es5.js?v=2.1.0-beta.6:321 – "Possibly unhandled rejection: {\"closeCode\":3001,\
    (anonymous function) (angular.js:11814)
    processChecks (angular.js:17943)
    $digest (angular.js:19074)
    $apply (angular.js:19462)
    (anonymous function) (angular.js:21563)
    completeTask (angular.js:21193)
    (anonymous function) (angular.js:6789)
```

Apparently while sending a message, the state changed.

The relevant code is in `sendChunk` (services/webclient.ts), a call to `this.relayedDataTask.sendMessage`. @lgrahl since the chunks should all be queued, can we simply log and disregard send errors?

**Comment by lgrahl** [comment]
This is the result of a timeout that fired after 1s:

1. A thumbnail has been requested (timeout scheduled).
2. The connection went away and has been closed (not able to resume).
3. The timeout fired as it was not cancelled by the closed session - this is a problem! This can not only lead to the above error but also to message requests leaking from one session into another.

Suggested solution:

Timeouts that send messages should be scheduled on the web client service. The web client service can then cancel all scheduled timeouts when the connection is being stopped and cannot be resumed.

**Comment by lgrahl** [comment]
There are a bunch of scheduled timeouts that follow the same behaviour. All of them will have to be updated.

**Comment by dbrgn** [comment]
:+1: thanks for the analysis! I'll try to tackle that tomorrow.

**Comment by ovalseven8** [comment]
Is this fixed? #620 says so.

**Comment by dbrgn** [comment]
Yes, thanks.
